crates/solvers/src/boundary/baseline.rs

```diff
@@ -44,34 +44,50 @@ impl<'a> Solver<'a> {
             max_hops,
         );
 
-        let (path, executed_sell_amount) = match request.side {
+        let segments = match request.side {
             order::Side::Buy => {
-                let best = candidates
+                let (segments, _) = candidates
                     .iter()
                     .filter_map(|path| {
-                        baseline_solver::estimate_sell_amount(request.buy.amount, path, &self.amms)
+                        let estimate = baseline_solver::estimate_sell_amount(
+                            request.buy.amount,
+                            path,
+                            &self.amms,
+                        )?;
+                        let segments = self.traverse_path(
+                            &estimate.path,
+                            request.sell.token.0,
+                            estimate.value,
+                        )?;
+
+                        let sell = estimate.value;
+                        let buy = segments
+                            .last()
+                            .map(|segment| segment.output.amount)
+                            .unwrap_or(sell);
+
+                        if sell <= request.sell.amount && buy >= request.buy.amount {
```

Comment:
Pointing out that `bool::then_some()` is a thing (although it might end up with bad formatting due to the long expression.

Also would be nice to add a small comment explaining why that is needed here so people don't have to track down the reason using `git blame`.

Reply:
> Pointing out that `bool::then_some()` is a thing

Thanks... I **always** forget about it (and agree that it generally makes nicer code).

If it is OK with you, I would not implement these suggestions (although I agree with them) just because the code changes again in https://github.com/cowprotocol/services/pull/1766.